lib/pure/streams.nim

```diff
@@ -338,8 +342,103 @@ proc peekLine*(s: Stream): TaintedString =
   defer: setPosition(s, pos)
   result = readLine(s)
 
+
+# copyMem is not available for js #5667
 when not defined(js):
 
+  proc unpackInt*(s: Stream, format: string): int =
```

Comment:
Ugh, no. Not another single character format language. Nim is not Python, you can do: 

``unpackInt("\x7f", uint, bigEndian)``

Reply:
Similar parsers exist in C, Perl and other languages. As a string, the formatter can be stored in const variables, arrays, tables and selected dynamically at runtime to implement protocols and file formats.

Reply:
So put it in a new module but the format language is cryptic and I doubt dynamic selection of these format strings is ever useful. C and Perl have weak support for compiletime features and so are not convincing examples.

Reply:
I'm going to agree with @Araq here. This should be done at compile-time as well.

We can offer support for the same format as Python/Perl/etc in a Nimble package I think (since each language likely has some oddities).